.github/workflows/update.yaml

```
@@ -1,26 +1,82 @@
-name: Update Lean
+name: Update and deploy
+# Everyday attempt to update Lean and mathlib. If the update succeeds then deploy the updated version to the webpage.
+
 on:
   schedule:
-    - cron: "0 11 * * 4"    # every Wednesday at 11:00 UTC
-  push:
-    branches:
-      - main
-  workflow_dispatch:
+    - cron: "0 8 * * *" # every day at 8:00 UTC
+  workflow_dispatch: # allows workflow to be triggered manually
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
-  update_lean:
+  update:
     runs-on: ubuntu-latest
     permissions:
-      issues: write         # required to create issues
-      pull-requests: write  # required to create pull requests
       contents: write
+    outputs:
+      updated: ${{ steps.update-tactic-list.outputs.files_changed }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
+
       - name: Update Lean project
-        uses:  oliver-butterley/lean-update-action@v1-alpha.3
+        id: update-lean-mathlib
+        uses: oliver-butterley/lean-update@v1-alpha
+
+      - name: When update succeeds generate markdown file
+        if: steps.update-lean-mathlib.outputs.result == 'update-success'
+        id: update-tactic-list
+        run: |
+          lake env lean --run Main.lean > src/tactics.md
+          python3 script.py
+          if [[ -n $(git diff -w "src/tactics.md") ]]; then
+            echo "files_changed=true"  >> "$GITHUB_OUTPUT"
+           else
+            echo "files_changed=false" >> "$GITHUB_OUTPUT"
+          fi
+          cat "$GITHUB_OUTPUT"
+
+      - name: Commit updated version
+        if: steps.update-lean-mathlib.outputs.result == 'update-success'
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR_ID+$GITHUB_ACTOR@users.noreply.github.com"
+          git add .
+          git commit -m "Updated version from cron job"
+          git push
+        env:
+          GITHUB_ACTOR: ${{ github.actor }}
+          GITHUB_ACTOR_ID: ${{ github.actor_id }}
+
+  deploy:
+    if: needs.update.outputs.updated == 'true' && github.ref == 'refs/heads/main'
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    needs: update
+    permissions:
+      pages: write
+      id-token: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup mdbook
+        uses: peaceiris/actions-mdbook@v2
         with:
-          #  Allowed values: "silent", "commit", "issue" or "pr". Default: "commit".
-          on_update_succeeds: pr
-          # Allowed values: "silent", "issue" or "fail". Default: "issue".
-          on_update_fails: issue
+          mdbook-version: "latest"
```

use the same version to `build.yaml`
```suggestion
          mdbook-version: '0.4.32'
```